### ISSUE TITLE
ps3eye_sdl.exe closes immediately in step 4

### ISSUE THREAD

**Issue by Stankiem** [brendanwalker/psmove-unity5#8]
I followed all the directions outlined in the wiki, however ps3eye_sdl.exe will not run, it simply closes immediately after opening.  I've been through the previous steps many times to make sure everything is correct and it is.  I did the 3 steps in the troubleshooting guide and did various other steps to try and get this working, such as reset the computer, switch USB ports multiple times, uninstalled reinstalled via USBDeviceView.  Really hoping for some help here.  Thanks so much for writing this all up and any help you can offer.  I am on Windows 10 btw

-Make sure Unity is shut down
-Make sure there are no other apps trying to connect to the camera
-If the red light next to the camera is still on, unplug and re-plug in the PS3Eye and then try running the apps again.

EDIT: Got it working!  The last set of USB ports on my PC worked for whatever reason...  But having another issue with the camera config files not existing.

**Comment by Stankiem** [comment]
I was able to get through step 5 just fine btw, until I got to test_tracker.exe obviously which didn't show either.

**Comment by Stankiem** [comment]
Got it working by using the last set of USB ports on my machine...  Dunno what makes them so different but it worked.

I am now having issues with the video jumping around while running test_tracking.exe however.  And I don't have the registry keys or files:  
HKEY_CURRENT_USER\Software\PS3EyeCamera
colormapping.dat
PSEye_backup.ini
PSEye_backup_win.ini

Why wouldn't these be built?

**Comment by brendanwalker** [comment]
Sorry for my slow reply. Was out of town most of this weekend. 

Those files (and registry key) will only be generated when using a version of the psmoveapi library with the CLEye driver (via the CAMERA_CONTROL_USE_CL_DRIVER option), see [here](https://github.com/brendanwalker/psmoveapi/blob/master/src/tracker/platform/camera_control_win32.c#L41-L81). I believe I didn't compile the version of psmoveapi used by psmove-unity5 in the master branch with that flag.

The camera issues tend to manifest on usb ports where there is a bunch of other USB traffic on the same usb root hub i.e. another camera (like the Oculus tracking camera). On the back of your computer pairs of usb ports are generally connected to the same hub. So you want to avoid putting a ps3 eye camera adjacent another camera like so:

[](https://cloud.githubusercontent.com/assets/3753542/11479853/ff09dbe0-9748-11e5-8614-01b0447a1e5f.jpg)

All that said, it turns out there were some inefficiencies in how the ps3eye camera driver worked that causes it to be more sensitive to other traffic than it should be. There is a large discussion about that [here](https://github.com/cboulay/psmove-ue4/issues/18). At the end of that discussion there is a potential camera driver fix I'm going to try out tonight. I'll update this issue with the results.

**Comment by Stankiem** [comment]
Thank you for the reply,  After reading some of your other replies regarding this problem I tried every combination of USB ports I could, disconnecting all other devices and putting keyboard and mouse in a completely seperate bank. etc..  Nothing helped.  There are only 2 USB ports where the device even works as described in your wiki.  I purchased this PCIe USB expansion card hoping to fix the issue:
 http://www.amazon.com/gp/product/B00FPIMJEW?psc=1&redirect=true&ref_=od_aui_detailpages00

That is arriving Thursday and I will let you know the results, but of course any progress you make on the driver front would be very welcome!

If there is anything else I can do to help you get to the bottom of these issues please let me know  =)  Once I get this working decently it will be extremely helpful!

**Comment by brendanwalker** [comment]
Thanks for linking that PCIe USB expansion port. I actually needed to order one of those myself but couldn't remember what one to people recommended. I think we're close on the driver. I was having similar variable framerate issues here in the ps3eye test app on my PC trying to run at 60fps, getting between 28 and 36MB/sec from the camera:

![image](https://cloud.githubusercontent.com/assets/3753542/12575414/0f611d88-c3c1-11e5-80e8-9878ec589673.png) 

Then after removing a sleep statement in the SDL example's [frame polling loop](https://github.com/brendanwalker/PS3EYEDriver/blob/rovarma-optimization/sdl/main.cpp#L206) I got a stable 36 MB/s:

![image](https://cloud.githubusercontent.com/assets/3753542/12575448/5d451964-c3c1-11e5-9dd7-5b176b68125b.png)

Hopefully I can get the same performance when using the psmoveapi in Unity.

**Comment by Stankiem** [comment]
Sounds like good news, I look forward to your next updates!  Great work, keep it up.

**Comment by brendanwalker** [comment]
I just got rovarma's libusb.ps3eye improvements incorporated into psmove-unity5. When you get a chance would you mind pulling down the following branch:

https://github.com/brendanwalker/psmove-unity5/tree/new_ps3driver

Here is a video of it in action in Unity for me: 
https://youtu.be/qI_lyUydsbc

And here is a video of the new version of ps3eye running at an (almost) solid 60fps:
https://youtu.be/2cVb4bHNKuQ

If this works for you I'll merge this back up into master

**Comment by Stankiem** [comment]
Thank you I will give it a try tonight!

**Comment by Stankiem** [comment]
I am having problems getting it to work at all now.  Below is a screenshot of my USBDeviceView and the error I am getting when I try and run PS3eye_SDL.exe
![psmove_error_ss](https://cloud.githubusercontent.com/assets/16855380/12687850/00df3c3e-c698-11e5-874b-4e5cb74fd5e7.png)

**Comment by brendanwalker** [comment]
Hmm. There shouldn't have been any other setup, the new build should have just worked. I just cloned https://github.com/brendanwalker/psmove-unity5/tree/new_ps3driver to my Win8.1 work machine and it ran ok. When I get home I can test it on my Win10 machine. In the mean time, some questions for you:

* Do you get the same problem when you run the old version of ps3eye_sdl in https://github.com/brendanwalker/psmove-unity5/tree/master? 
* Are you running this using an admin account or a non-admin account. Many of the tools have issues opening USB devices when not run on an admin account.
* Does the test_tracker.exe app also crash on startup or does run ok?

**Comment by Stankiem** [comment]
1- When I run the old version of ps3eye_sdl it simply closes immediately after I start it without any error message, it doesn't give me any error.
2- Yes I am using an admin account, doing run as administrator also.
3- The test_tracker.exe app closes immediately after opening it also in both the master and this branch.

Just a couple more things that may help..
-If I uninstall the device and then run the ps3eye_sdk.exe without first switching the camera (Interface 0) driver to the libusb-win32 driver it actually runs,  On the old version I get 180fps and it's jumpy, on the new it's like 20fps and insanely choppy.  Not sure what this means, if anything??

-When I was first following your directions I think I switched the camera (interface 1) to libusb-win32 instead of (Interface 0), not sure if this would matter..  But in your screenshot I see your camera interface 1 service names are usbaudio, in mine it is WinUSB.

Thanks so much for all this!!!

**Comment by Stankiem** [comment]
IT'S WORKING!  Not exactly sure what I did, I just went through task manager, uninstalled all USB drivers I could find besides the keyboard and mouse, uninstalled all USB audio devices, DK2, etc.. etc.. etc.. and checked the "delete driver software" box.  Restarted the computer, plugged it in the new USB card, then everything worked as expected.

FYI ALL - the test_tracker program will close immediately if the pS move controllers are not on.  I think that is why when I had the wrong drivers installed above, the ps3eye_sdl sorta worked and test tracker didn't... just a quick note on that since it scared me for a moment that it didn't work after the ps3eye_sdl worked so well.

It seems to be working near perfectly now.  Thanks again for all your help.

**Comment by brendanwalker** [comment]
Sweet! Thanks for the update. In Win10 I have occasionally been bitten by stale USB drivers on the camera. Completely removing the USB drivers and starting from a clean slate usually does the trick.

And yeah I could have a more friendly warning when the controller is turned off. This will get better once we get the new PSMoveService working, but that's still at least a month out.

There are still a few small tweaks I'm working with rovarma on the ps3eye_sdl app, but once we have those settled I'll migrate these fixes back up to master. Once that is done I'll close this issue.
